test/tools/yulInterpreter/EwasmBuiltinInterpreter.cpp

```diff
@@ -335,38 +335,38 @@ u256 EwasmBuiltinInterpreter::evalEthBuiltin(string const& _fun, vector<uint64_t
 		return m_state.calldata.size();
 	else if (_fun == "callCode")
 	{
-		// TODO read args from memory
-		// TODO use readAddress to read address.
+		readAddress(arg[1]);
+		readU128(arg[2]);
+		accessMemory(arg[3], arg[4]);
 		logTrace(evmasm::Instruction::CALLCODE, {});
 		return arg[0] & 1;
 	}
 	else if (_fun == "callDelegate")
 	{
-		// TODO read args from memory
-		// TODO use readAddress to read address.
+		readAddress(arg[1]);
+		accessMemory(arg[2], arg[3]);
 		logTrace(evmasm::Instruction::DELEGATECALL, {});
 		return arg[0] & 1;
 	}
 	else if (_fun == "callStatic")
 	{
-		// TODO read args from memory
-		// TODO use readAddress to read address.
+		readAddress(arg[1]);
+		accessMemory(arg[2], arg[3]);
 		logTrace(evmasm::Instruction::STATICCALL, {});
 		return arg[0] & 1;
 	}
 	else if (_fun == "storageStore")
 	{
-		m_state.storage[h256(readU256(arg[0]))] = readU256((arg[1]));
+		m_state.storage[readBytes32(arg[0])] = readBytes32(arg[1]);
 		return 0;
 	}
 	else if (_fun == "storageLoad")
 	{
-		writeU256(arg[1], m_state.storage[h256(readU256(arg[0]))]);
+		writeBytes32(arg[1], m_state.storage[readBytes32(arg[0])]);
```

Comment:
maybe we should also call `accessMemory` here and in `storageStore` ?

Reply:
`writeBytes32` does that already

Reply:
ah great :)